crypto/evp/digest.c

```diff
@@ -678,9 +678,14 @@ const OSSL_PARAM *EVP_MD_CTX_settable_params(EVP_MD_CTX *ctx)
         return pctx->op.sig.signature->settable_ctx_md_params(
                    pctx->op.sig.sigprovctx);
 
-    if (ctx->digest != NULL && ctx->digest->settable_ctx_params != NULL)
-        return ctx->digest->settable_ctx_params(
-                  ossl_provider_ctx(EVP_MD_provider(ctx->digest)));
+    if (ctx->digest != NULL) {
+        void *prov_ctx = ossl_provider_ctx(EVP_MD_provider(ctx->digest));
+
+        if (ctx->digest->ctx_settable_params != NULL)
+            return ctx->digest->ctx_settable_params(prov_ctx, ctx->provctx);
```

Comment:
In hindsight, `ctx->provctx` isn't the best name...

Reply:
Agreed.